.github/workflows/python-ci.yml

```diff
@@ -24,6 +24,24 @@ on:
     branches:
     - 'main'
   pull_request:
+    paths-ignore:
+    - '.github/ISSUE_TEMPLATE/**'
+    - '.github/workflows/check-md-link.yml'
+    - '.github/workflows/license_check.yml'
+    - '.github/workflows/python-ci-docs.yml'
+    - '.github/workflows/python-integration.yml'
+    - '.github/workflows/python-release.yml'
+    - '.github/workflows/python-release-docs.yml'
+    - '.github/workflows/stale.yml'
+    - '.gitignore'
+    - '.asf.yml'
+    - 'dev/**'
```

Comment:
all the other paths LGTM except `dev/**`. This is where we set up the docker env for integration tests. Since this GitHub Action runs the integration tests, we should not ignore when `dev/**` changes.